placeholders_test.go

```diff
@@ -0,0 +1,132 @@
+package txtdirect
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestParsePlaceholders(t *testing.T) {
+	tests := []struct {
+		url       string
+		requested string
+		expected  string
+	}{
+		{
+			"example.com{uri}",
+			"https://example.com/?test=test",
+			"example.com/?test=test",
+		},
+		{
+			"example.com{uri}/{~test}",
+			"https://example.com/?test=test",
+			"example.com/?test=test/test",
+		},
+		{
+			"example.com{uri}/{>Test}",
+			"https://example.com/?test=test",
+			"example.com/?test=test/test-header",
+		},
+		{
+			"example.com{uri}/{?test}",
+			"https://example.com/?test=test",
+			"example.com/?test=test/test",
+		},
+		{
+			"example.com{dir}",
+			"https://example.com/directory/test",
+			"example.com/directory/",
+		},
+		{
+			"example.com/directory/{file}",
+			"https://example.com/directory/file.pdf",
+			"example.com/directory/file.pdf",
+		},
+		{
+			"example.com/?test={fragment}",
+			"https://example.com/?test=#fragment",
+			"example.com/?test=fragment",
+		},
+		{
+			"example.com/{host}",
+			"https://project.example.com:8080",
+			"example.com/project.example.com:8080",
+		},
+		{
+			"example.com/{hostonly}",
+			"https://project.example.com",
+			"example.com/project.example.com",
+		},
+		{
+			"example.com/{method}",
+			"https://example.com",
+			"example.com/GET",
+		},
+		{
+			"example.com{path}",
```

Comment:
This should probably test a longer or more complicated path.

Reply:
I added a more complex path example. Happy to make additional changes if you have a different example in mind.